timemachine/fe/free_energy.py

```diff
@@ -154,11 +155,35 @@ class InitialState:
     lamb: float
     ligand_idxs: NDArray
     protein_idxs: NDArray
+    atom_map: Optional[AtomMapMixin] = None
 
     def __post_init__(self):
         assert self.ligand_idxs.dtype == np.int32 or self.ligand_idxs.dtype == np.int64
         assert self.protein_idxs.dtype == np.int32 or self.protein_idxs.dtype == np.int64
 
+    def get_interacting_ligand_atom_indices(self) -> NDArray:
+        """Returns the ligand atom indices that are fully interacting at the specific lambda.
+
+        Currently implemented by looking at the atoms in the atom_map object and either return
+        the endstates for lambda == 0.0 and lambda == 1.0 and for all other values the core atoms.
+
+        Note
+        ----
+        * Raises an exception if no atom map object is provided
+
+        Return
+        ------
+        np.ndarray
+            Contains the indices of the ligand that are not considered dummy atoms at the lambda value.
+        """
+        assert self.atom_map is not None, "No atom map provided, unable to provide interacting atom indices"
+        if self.lamb == 0.0:
+            return self.ligand_idxs[self.atom_map.c_flags != 2]
```

Comment:
nit: It'd be nice if these constants were defined somewhere

Reply:
Added an enum that defines the types in https://github.com/proteneer/timemachine/pull/1338/commits/4891138630208cbace3fddb8b9cbe4070160972d